sdk/communication/communication-chat/review/communication-chat.api.md

```diff
@@ -76,50 +76,69 @@ export interface ChatClientOptions extends PipelineOptions {
 }
 
 // @public
-export interface ChatMessage extends Omit<RestChatMessage, "senderId" | "content"> {
+export interface ChatMessage {
     content?: ChatMessageContent;
-    sender?: CommunicationUserIdentifier;
+    createdOn: Date;
+    deletedOn?: Date;
+    editedOn?: Date;
+    id: string;
+    sender?: CommunicationIdentifier;
```

Comment:
This needs to be a `CommunicationIdentifierKind`. In TS we use discriminated unions so that developers can switch-case on the `kind` property.

The rule is: all identifiers that get passed as input to the SDK need to be the type without kind, and all identifiers that get returned from the SDK need to include the kind.

Reply:
fixed